### PR TITLE
ai-chat: make tool confirmation default a dedicated preference

### PR DESCRIPTION
**Pull request by sgraband**
#### What it does
Replace the '*' magic-key inside ai-features.chat.toolConfirmation with a dedicated preference ai-features.chat.defaultToolConfirmation. The new preference is surfaced in the standard settings UI and defaults to 'confirm' instead of 'always_allow', so tools require user approval unless explicitly opted in.

- Add ai-features.chat.defaultToolConfirmation (string enum, default 'confirm') alongside the existing per-tool record preference.
- Extend ToolConfirmationManager with getDefaultConfirmationMode and setDefaultConfirmationMode; drop the '*' branch from getConfirmationMode and reset behaviour. Read the new preference via TrustAwarePreferenceReader to keep the workspace-trust hardening.
- Preserve confirmAlwaysAllow semantics: such tools default to 'confirm', never inherit 'always_allow' from the global default; explicit per-tool entries are still respected.
- Update the tools configuration widget to use the new APIs and listen for changes on the new preference.
- Deduplicate the schema by sharing enum values and descriptions between the two preferences; remove the leftover 'yolo' NLS key.
- Rewrite the unit tests to cover the new public API surface, workspace-trust on the new preference, and the confirmAlwaysAllow edge cases.

#### How to test
- Open the settings.json, settings and AI config view (Tools Tab) side by side and manipulate the relevant settings in all ways possible. Check that everything stays up to date.
- Trigger some functions from the chat to see that the correct confirmation mode is active.

#### Breaking changes

- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)
- [x] User-facing text is internationalized using the `nls` service (for details, please see the [Internationalization/Localization section](https://github.com/theia-ide/theia/blob/master/doc/coding-guidelines.md#internationalizationlocalization) in the Coding Guidelines)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat/src/browser/chat-tool-preference-bindings.ts [hunk 1]
```diff
@@ -36,12 +40,30 @@ export class ToolConfirmationManager {
...
+    getDefaultConfirmationMode(): ToolConfirmationMode {
+        const value = this.trustAwareReader.get<ToolConfirmationMode>(DEFAULT_TOOL_CONFIRMATION_PREFERENCE);
+        return value ?? this.getDefaultPreferenceSchemaDefault();
+    }
+
+    /**
+     * Set the global default confirmation mode.
+     */
+    setDefaultConfirmationMode(mode: ToolConfirmationMode): void {
+        this.preferenceService.updateValue(DEFAULT_TOOL_CONFIRMATION_PREFERENCE, mode);
```
Comment (eneufeld) [inline packages/ai-chat/src/browser/chat-tool-preference-bindings.ts]: this is async, should we make the set also async? somebody should handle the error case here maybe? see packages/ai-ide/src/browser/ai-configuration/tools-configuration-widget.tsx#136

**Review by eneufeld — COMMENTED** [review]
I would claim we should mark this as breaking:
`"ai-features.chat.toolConfirmation": {
  "*": "always_allow"
}`
this is ignored now.
We should add some info about this to the release imho

**Review by eneufeld — APPROVED** [review]
LGTM. thank you!
